gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+    List<String> cpuPlatforms = new ArrayList<String>();
+    zone = nameFromSelfLink(zone);
+    Zone zoneObject = compute.zones().get(projectId, zone).execute();
+    if (zoneObject == null) {
+      return cpuPlatforms;
+    }
+    return zoneObject.getAvailableCpuPlatforms();
+  }
+
+  public List<DiskType> getDiskTypes(String projectId, String zone) throws IOException {
+    zone = nameFromSelfLink(zone);
+    List<DiskType> diskTypes = compute.diskTypes().list(projectId, zone).execute().getItems();
+    if (diskTypes == null) {
+      return new ArrayList<>();
+    }
+
+    // No deprecated items
+    Iterator it = diskTypes.iterator();
+    while (it.hasNext()) {
+      DiskType o = (DiskType) it.next();
+      if (o.getDeprecated() != null
+          && o.getDeprecated().getState().equalsIgnoreCase("DEPRECATED")) {
+        it.remove();
+      }
+    }
+
+    diskTypes.sort(Comparator.comparing(DiskType::getName));
+
+    return diskTypes;
+  }
+
+  public List<DiskType> getBootDiskTypes(String projectId, String zone) throws IOException {
+    zone = nameFromSelfLink(zone);
+    List<DiskType> diskTypes = this.getDiskTypes(projectId, zone);
+
+    // No local disks
+    Iterator it = diskTypes.iterator();
+    while (it.hasNext()) {
+      DiskType o = (DiskType) it.next();
+      if (o.getName().startsWith("local-")) {
+        it.remove();
+      }
+    }
+    return diskTypes;
+  }
+
+  public List<Image> getImages(String projectId) throws IOException {
+    List<Image> images = compute.images().list(projectId).execute().getItems();
+    if (images == null) {
+      return new ArrayList<>();
+    }
+
+    // No deprecated items
+    Iterator it = images.iterator();
+    while (it.hasNext()) {
+      Image o = (Image) it.next();
+      if (o.getDeprecated() != null
+          && o.getDeprecated().getState().equalsIgnoreCase("DEPRECATED")) {
+        it.remove();
+      }
+    }
+
+    images.sort(Comparator.comparing(Image::getName));
+
+    return images;
+  }
+
+  public Image getImage(String projectId, String name) throws IOException {
+    Image image = compute.images().get(projectId, name).execute();
+
+    return image;
+  }
+
+  public List<AcceleratorType> getAcceleratorTypes(String projectId, String zone)
+      throws IOException {
+    zone = nameFromSelfLink(zone);
+
+    List<AcceleratorType> acceleratorTypes =
+        compute.acceleratorTypes().list(projectId, zone).execute().getItems();
+    if (acceleratorTypes == null) {
+      return new ArrayList<>();
+    }
+
+    // No deprecated items
+    Iterator it = acceleratorTypes.iterator();
+    while (it.hasNext()) {
+      AcceleratorType o = (AcceleratorType) it.next();
+      if (o.getDeprecated() != null
+          && o.getDeprecated().getState().equalsIgnoreCase("DEPRECATED")) {
+        it.remove();
+      }
+    }
+    acceleratorTypes.sort(Comparator.comparing(AcceleratorType::getName));
+    return acceleratorTypes;
```

Comment:
It looks like this pattern is pretty common. It might be worth using an adapter to de-duplciate the logic.

Reply:
Done with ClientUtil::processResourceList